Shared/Samples/Find route around barriers/FindRouteAroundBarriersView.swift

```diff
@@ -148,6 +142,97 @@ struct FindRouteAroundBarriersView: View {
+                    }
+                }
+            }
+        }
+    }
+    
+    @ViewBuilder var settings: some View {
+        NavigationStack {
+            Form {
+                Section("Features") {
+                    Picker("Features", selection: $featuresSelection) {
+                        Text("Stops").tag(RouteFeatures.stops)
+                        Text("Barriers").tag(RouteFeatures.barriers)
+                    }
+                    .pickerStyle(.segmented)
+                }
+                RouteParametersSettings(for: model.routeParameters)
+            }
+            .navigationTitle("Settings")
+            .navigationBarTitleDisplayMode(.inline)
+            .toolbar {
+                ToolbarItem(placement: .confirmationAction) {
+                    Button("Done") {
+                        settingsAreVisible = false
+                    }
+                }
+            }
+        }
+    }
+}
+
+extension FindRouteAroundBarriersView {
+    /// A list of settings for modifying route parameters.
+    struct RouteParametersSettings: View {
+        /// The route parameters to modify.
+        private let routeParameters: RouteParameters
+        
+        /// A Boolean value indicating whether routing will find the best sequence.
+        @State private var routingFindsBestSequence: Bool = true
+        
+        /// A Boolean value indicating whether routing will preserve the first stop.
+        @State private var routePreservesFirstStop: Bool = true
+        
+        /// A Boolean value indicating whether routing will preserve the last stop.
+        @State private var routePreservesLastStop: Bool = true
+        
+        init(for routeParameters: RouteParameters) {
+            self.routeParameters = routeParameters
+        }
+        
+        var body: some View {
+            Section {
+                Toggle("Find Best Sequence", isOn: $routingFindsBestSequence)
+                    .onChange(of: routingFindsBestSequence) {
+                        routeParameters.findsBestSequence = routingFindsBestSequence
+                    }
+            }
+            Section("Preserve Stops") {
+                Toggle("Preserve First Stop", isOn: $routePreservesFirstStop)
+                    .onChange(of: routePreservesFirstStop) {
+                        routeParameters.preservesFirstStop = routePreservesFirstStop
+                    }
+                Toggle("Preserve Last Stop", isOn: $routePreservesLastStop)
+                    .onChange(of: routePreservesLastStop) {
+                        routeParameters.preservesLastStop = routePreservesLastStop
+                    }
+            }
+            .disabled(!routingFindsBestSequence)
+            .onAppear {
+                self.routingFindsBestSequence = routeParameters.findsBestSequence
+                self.routePreservesFirstStop = routeParameters.preservesFirstStop
+                self.routePreservesLastStop = routeParameters.preservesLastStop
```

Comment:
`self.` isn't needed here.

```suggestion
                routingFindsBestSequence = routeParameters.findsBestSequence
                routePreservesFirstStop = routeParameters.preservesFirstStop
                routePreservesLastStop = routeParameters.preservesLastStop
```